common/libraries/crypto/bip32.c

```diff
@@ -757,75 +809,65 @@ int hdnode_nem_decrypt(const HDNode *node, const ed25519_public_key public_key,
+                      node->private_key,
+                      msg,
+                      msg_len,
+                      sig,
+                      pby,
+                      is_canonical);
   } else if (node->curve == &curve25519_info) {
-    return 1;  // signatures are not supported
+    return 1;    // signatures are not supported
   } else {
     if (node->curve == &ed25519_info) {
       hdnode_fill_public_key(node);
       ed25519_sign(msg, msg_len, node->private_key, node->public_key + 1, sig);
     } else if (node->curve == &ed25519_sha3_info) {
       hdnode_fill_public_key(node);
-      ed25519_sign_sha3(msg, msg_len, node->private_key, node->public_key + 1,
-                        sig);
+      ed25519_sign_sha3(
+          msg, msg_len, node->private_key, node->public_key + 1, sig);
 #if USE_KECCAK
     } else if (node->curve == &ed25519_keccak_info) {
       hdnode_fill_public_key(node);
-      ed25519_sign_keccak(msg, msg_len, node->private_key, node->public_key + 1,
-                          sig);
+      ed25519_sign_keccak(
+          msg, msg_len, node->private_key, node->public_key + 1, sig);
 #endif
     } else {
-      return 1;  // unknown or unsupported curve
+      return 1;    // unknown or unsupported curve
     }
     return 0;
   }
 }
 
-int hdnode_sign_digest(HDNode *node, const uint8_t *digest, uint8_t *sig,
+int hdnode_sign_digest(HDNode *node,
+                       const uint8_t *digest,
+                       uint8_t *sig,
                        uint8_t *pby,
                        int (*is_canonical)(uint8_t by, uint8_t sig[64])) {
   if (node->curve->params) {
-    return ecdsa_sign_digest(node->curve->params, node->private_key, digest,
-                             sig, pby, is_canonical);
+    return ecdsa_sign_digest(
+        node->curve->params, node->private_key, digest, sig, pby, is_canonical);
   } else if (node->curve == &curve25519_info) {
-    return 1;  // signatures are not supported
+    return 1;    // signatures are not supported
   } else {
     return hdnode_sign(node, digest, 32, 0, sig, pby, is_canonical);
   }
 }
 
-int hdnode_get_shared_key(const HDNode *node, const uint8_t *peer_public_key,
-                          uint8_t *session_key, int *result_size) {
-  // Use elliptic curve Diffie-Helman to compute shared session key
-  if (node->curve->params) {
-    if (ecdh_multiply(node->curve->params, node->private_key, peer_public_key,
-                      session_key) != 0) {
-      return 1;
-    }
-    *result_size = 65;
-    return 0;
-  } else if (node->curve == &curve25519_info) {
-    session_key[0] = 0x04;
-    if (peer_public_key[0] != 0x40) {
-      return 1;  // Curve25519 public key should start with 0x40 byte.
-    }
-    curve25519_scalarmult(session_key + 1, node->private_key,
-                          peer_public_key + 1);
-    *result_size = 33;
-    return 0;
-  } else {
-    *result_size = 0;
-    return 1;  // ECDH is not supported
-  }
-}
```

Comment:
is this delete intentional? why is this deleted?

Reply:
Not used anywhere, seems like an old function which was never used.